app/src/main/java/com/alphawallet/app/widget/TokenWithBalanceView.java

```diff
@@ -0,0 +1,49 @@
+package com.alphawallet.app.widget;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import androidx.annotation.Nullable;
+import com.alphawallet.app.R;
+import com.alphawallet.app.entity.tokens.Token;
+import timber.log.Timber;
+
+public class TokenWithBalanceView extends LinearLayout
+{
+    TokenIcon tokenIcon;
+    TextView tokenPrice;
+
+    public TokenWithBalanceView(Context context, @Nullable AttributeSet attrs)
+    {
+        super(context, attrs);
+        inflate(context, R.layout.item_token_with_balance_view, this);
+        tokenIcon = findViewById(R.id.token_icon);
+        tokenPrice = findViewById(R.id.title_set_price);
+    }
+
```

Comment:
There should be a method in here that populates the internal adapter from a ```Token[]``` array.